sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfigurationprovider/_azureappconfigurationprovider.py

```diff
@@ -0,0 +1,180 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+
+from urllib.parse import urlparse
+import json
+import warnings
+from azure.appconfiguration import AzureAppConfigurationClient
+from azure.keyvault.secrets import SecretClient
+from ._settingselector import SettingSelector
+from ._user_agent import USER_AGENT
+
+
+class AzureAppConfigurationProvider:
+    """
+    Provides a dictionary-like interface to Azure App Configuration settings. Enables loading of sets of configuration
+    settings from Azure App Configuration into a Python application. Enables trimming of prefixes from configuration
+    keys. Enables resolution of Key Vault references in configuration settings.
+    """
+
+    def __init__(self):
+        # type: () -> None
+        self._dict = {}
+        self._trim_prefixes = []
+        self._client = None
+
+    @classmethod
+    def load(cls, connection_string=None, endpoint=None, credential=None, **kwargs):
+        """
+        Loads configuration settings from Azure App Configuration into a Python application.
+        :param connection_string: Connection string (one of connection_string or endpoint and credential must be set)
+        :type connection_string: str
+        :param endpoint: Endpoint (one of connection_string or endpoint and credential must be set)
+        :type endpoint: str
+        :param credential: Credential (one of connection_string or endpoint and credential must be set)
+        :type credential: Union[AppConfigConnectionStringCredential, TokenCredential]
+        :keyword selectors: List of setting selectors to filter configuration settings
+        :type selectors: list[~azure.appconfigurationprovider.SettingSelector]
+        :keyword trim_prefixes: List of prefixes to trim from configuration keys
+        :type trim_prefixes: list[str]
+        :keyword key_vault_options: Options for resolving Key Vault references
+        :type key_vault_options: ~azure.appconfigurationprovider.KeyVaultOptions
+        """
+        provider = AzureAppConfigurationProvider()
+
+        key_vault_options = kwargs.pop("key_vault_options", None)
```

Comment:
Can `kwargs.get("key_vault_options")` be used instead since it defaults to `None` if the key is not found? Also, is there a reason why you are removing the keyword arguments from the dictionary ?

Reply:
I choose pop because, 1. we don't need it anywhere else and shouldn't be used anywhere else. 2. It is done other locations in the SDK this way. 3. I personally find it clearer that I intend None as the option.